submodules/wizard/wizard.js

```diff
@@ -1050,24 +1057,155 @@ define(function(require) {
 
 		/* APP RESTRICTIONS */
 
+		/**
+		 * Render App Restrictions step
+		 * @param  {Object} args
+		 * @param  {Object} args.data  Wizard's data that is shared across steps
+		 * @param  {Object} args.data.appRestrictions  Data specific for the current step
+		 * @param  {jQuery} args.container  Step container element
+		 */
 		wizardAppRestrictionsRender: function(args) {
 			var self = this,
-				$container = args.container;
+				$container = args.container,
+				initTemplate = function(appList) {
+					var appRestrictionsData = args.data.appRestrictions,
+						$template = $(self.getTemplate({
+							name: 'step-appRestrictions',
+							data: {
+								apps: appList,
+								anyAppToSelect: (appList.length - appRestrictionsData.allowedAppIds.length) > 0,
```

Comment:
It seems possible for a TypeError to be thrown here: if a user selects app access restriction but does not add any, they still are able to move to the next step.

If they come back to this step, `appRestrictionsData.allowedAppIds` would be undefined and an error will be thrown.